<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>AEngine_EditorClass</class>
 <widget class="QMainWindow" name="AEngine_EditorClass">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1097</width>
    <height>814</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Ignored" vsizetype="Ignored">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="cursor">
   <cursorShape>ArrowCursor</cursorShape>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::NoContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>Arenal Engine Editor</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow, QWidget, QDialog, QMenuBar::item
{
	background-color: rgb(50, 50, 50);
	color: rgb(255, 255, 255);
}

QHeaderView::section
{
	background-color: qlineargradient(spread:reflect, x1:0, y1:0, x2:0, y2:0.5, stop:0 rgba(50, 50, 50, 255), stop:1 rgba(75, 75, 75, 255));
	padding-left: 4px;
	border: 1px solid rgb(110, 110, 110);
}

QMenuBar::item:selected, QMenu::item:selected
{
    background-color: rgb(120, 120, 120);
	border: 1px solid rgb(200, 200, 200);
}

QPushButton:hover
{
    background-color: rgb(75, 75, 75);
}

QPushButton:pressed
{
    background-color: rgb(100, 100, 100);
}

QLineEdit
{
	border-width: 2px;
	border-style: inset;
	border-color: rgb(60, 60, 60);
	border-radius: 5px;
	padding: 1px;
}

QLineEdit:focus
{
	background-color: rgb(75, 75, 75);
}

QDockWidget::title
{
	background-color: rgb(65, 65, 65);
	padding: 5px;
}

QTabBar::tab 
{
	background-color: qlineargradient(spread:pad, x1:1, y1:0.5, x2:0, y2:0.5, stop:0 rgba(60, 60, 60, 255), stop:0.5 rgba(80, 80, 80, 255), stop:1 rgba(60, 60, 60, 255));
	padding: 5px;
	borde-width: 3px; 
	border-color: rgb(110, 110, 110);
	border-style: inset;
	border-radius: 8px;
	font-size: 11px;
}

QTabBar::tab:selected 
{
	background-color: qlineargradient(spread:pad, x1:1, y1:0.5, x2:0, y2:0.5, stop:0 rgba(60, 60, 60, 255), stop:0.5 rgba(120, 120, 120, 255), stop:1 rgba(60, 60, 60, 255));
	border-style: outset;
    border-bottom-color: rgb(60, 60, 60);
}

QTreeWidget
{
	alternate-background-color: rgb(60, 60, 60);
}

QTreeWidget::branch:has-children
{
	image: url(:/AEngine_Editor/Resources/Images/branch-closegrey.png) 0;
}

QTreeWidget::branch:has-children:open
{
	image: url(:/AEngine_Editor/Resources/Images/branch-opengrey.png) 0;
}

QTreeWidget::branch:has-children:open:selected
{
	image: url(:/AEngine_Editor/Resources/Images/branch-openblue.png) 0;
}

QTreeWidget::branch:has-children:!open:selected
{
	image: url(:/AEngine_Editor/Resources/Images/branch-closeblue.png) 0;
}
</string>
  </property>
  <property name="documentMode">
   <bool>true</bool>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks|QMainWindow::VerticalTabs</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="mouseTracking">
    <bool>false</bool>
   </property>
   <property name="focusPolicy">
    <enum>Qt::NoFocus</enum>
   </property>
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QMenuBar" name="m_MainMenu">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1097</width>
     <height>21</height>
    </rect>
   </property>
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <widget class="QMenu" name="m_MainMenuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="m_MenuFileCreateProjectAction"/>
    <addaction name="m_MenuFileLoadProjectAction"/>
    <addaction name="m_MenuFileSaveProjectAction"/>
    <addaction name="separator"/>
    <addaction name="m_MenuFileExitAction"/>
   </widget>
   <widget class="QMenu" name="m_MenuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="m_RawGameAssetsViewAction"/>
    <addaction name="m_GameAssetsViewAction"/>
    <addaction name="m_GameObjectsViewAction"/>
    <addaction name="m_GameObjectPropertiesViewAction"/>
    <addaction name="separator"/>
    <addaction name="m_RefreshAllViewsAction"/>
    <addaction name="separator"/>
   </widget>
   <widget class="QMenu" name="m_MenuProject">
    <property name="title">
     <string>Project</string>
    </property>
   </widget>
   <widget class="QMenu" name="m_MenuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <widget class="QMenu" name="menuPhysics">
     <property name="title">
      <string>Physics</string>
     </property>
     <addaction name="m_StartPhysXDebuggerAction"/>
     <addaction name="m_StopPhysXDebuggerAction"/>
     <addaction name="separator"/>
    </widget>
    <addaction name="m_SaveLayoutAction"/>
    <addaction name="m_LoadLayoutAction"/>
    <addaction name="separator"/>
    <addaction name="menuPhysics"/>
   </widget>
   <widget class="QMenu" name="m_MenuGameObjects">
    <property name="title">
     <string>Game Objects</string>
    </property>
    <widget class="QMenu" name="m_MenuCreateGameObjects">
     <property name="title">
      <string>Create Game Object</string>
     </property>
     <addaction name="m_CreateEmptyGameObjectAction"/>
    </widget>
    <addaction name="m_MenuCreateGameObjects"/>
   </widget>
   <widget class="QMenu" name="m_MenuGameAssets">
    <property name="title">
     <string>Game Assets</string>
    </property>
    <widget class="QMenu" name="m_MenuBuiltInGameAssets">
     <property name="title">
      <string>Built-in Game Assets</string>
     </property>
     <widget class="QMenu" name="m_MenuBIShaders">
      <property name="title">
       <string>Shaders</string>
      </property>
      <addaction name="m_BIDiffuseTextureMaterialVSAction"/>
      <addaction name="m_BIDiffuseTextureMaterialPSAction"/>
     </widget>
     <addaction name="m_MenuBIShaders"/>
    </widget>
    <addaction name="m_NewRawGameAssetAction"/>
    <addaction name="m_ReloadOutdatedRawGameAssetsAction"/>
    <addaction name="separator"/>
    <addaction name="m_MenuBuiltInGameAssets"/>
   </widget>
   <addaction name="m_MainMenuFile"/>
   <addaction name="m_MenuProject"/>
   <addaction name="m_MenuGameAssets"/>
   <addaction name="m_MenuGameObjects"/>
   <addaction name="m_MenuView"/>
   <addaction name="m_MenuSettings"/>
  </widget>
  <widget class="QDockWidget" name="m_RawGameAssetsDock">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="acceptDrops">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Raw Game Assets</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="m_RawGameAssetsDockContent">
    <property name="contextMenuPolicy">
     <enum>Qt::NoContextMenu</enum>
    </property>
    <property name="acceptDrops">
     <bool>false</bool>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="RawGameAssetTreeWidget" name="m_RawGameAssetsTree">
       <property name="acceptDrops">
        <bool>true</bool>
       </property>
       <property name="dragDropOverwriteMode">
        <bool>true</bool>
       </property>
       <property name="dragDropMode">
        <enum>QAbstractItemView::DropOnly</enum>
       </property>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="m_GameAssetsDock">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>Game Assets</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="m_GameAssetsDockContent">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="GameAssetTreeWidget" name="m_GameAssetsTree">
       <property name="dragDropMode">
        <enum>QAbstractItemView::DragOnly</enum>
       </property>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="m_GameObjectsDock">
   <property name="contextMenuPolicy">
    <enum>Qt::NoContextMenu</enum>
   </property>
   <property name="windowTitle">
    <string>Game Objects</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="m_GameObjectsDockContainter">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="GameObjectTreeWidget" name="m_GameObjectsTree">
       <property name="indentation">
        <number>10</number>
       </property>
       <property name="animated">
        <bool>true</bool>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="headerDefaultSectionSize">
        <number>80</number>
       </attribute>
       <attribute name="headerMinimumSectionSize">
        <number>10</number>
       </attribute>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="m_GameObjectPropsDock">
   <property name="windowTitle">
    <string>Game Object Properties</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="m_GameObjectPropsDockContainer">
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="GameObjectPropsTreeWidget" name="m_GameObjectPropsTree">
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true"/>
       </property>
       <property name="alternatingRowColors">
        <bool>false</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::NoSelection</enum>
       </property>
       <property name="indentation">
        <number>0</number>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string notr="true">1</string>
        </property>
       </column>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="m_HLayoutGOProps" stretch="0,0">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
       <item>
        <spacer name="m_HSpacerGOPropsAddComponent">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>130</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="m_AddGOComponentButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Add Component</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="m_ToolBar">
   <property name="windowTitle">
    <string>Tool Bar</string>
   </property>
   <property name="layoutDirection">
    <enum>Qt::LeftToRight</enum>
   </property>
   <property name="allowedAreas">
    <set>Qt::AllToolBarAreas</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="separator"/>
   <addaction name="m_PlayAction"/>
   <addaction name="m_PauseAction"/>
   <addaction name="m_StopAction"/>
   <addaction name="separator"/>
  </widget>
  <widget class="QStatusBar" name="m_StatusBar"/>
  <action name="m_MenuFileExitAction">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="m_RawGameAssetsViewAction">
   <property name="text">
    <string>Raw Game Assets</string>
   </property>
  </action>
  <action name="m_GameAssetsViewAction">
   <property name="text">
    <string>Game Assets</string>
   </property>
  </action>
  <action name="m_GameObjectsViewAction">
   <property name="text">
    <string>Game Objects</string>
   </property>
  </action>
  <action name="m_SaveLayoutAction">
   <property name="text">
    <string>Save Layout</string>
   </property>
  </action>
  <action name="m_LoadLayoutAction">
   <property name="text">
    <string>Load Layout</string>
   </property>
  </action>
  <action name="m_NewRawGameAssetAction">
   <property name="text">
    <string>New Raw Game Asset</string>
   </property>
  </action>
  <action name="m_ReloadOutdatedRawGameAssetsAction">
   <property name="text">
    <string>Reload outdated Raw Game Assets</string>
   </property>
  </action>
  <action name="m_CreateEmptyGameObjectAction">
   <property name="text">
    <string>Empty Game Object</string>
   </property>
  </action>
  <action name="m_GameObjectPropertiesViewAction">
   <property name="text">
    <string>Game Object Properties</string>
   </property>
  </action>
  <action name="m_BIDiffuseTextureMaterialPSAction">
   <property name="text">
    <string>Diffuse Texture Material PS</string>
   </property>
  </action>
  <action name="m_BIDiffuseTextureMaterialVSAction">
   <property name="text">
    <string>Diffuse Texture Material VS</string>
   </property>
  </action>
  <action name="m_RefreshAllViewsAction">
   <property name="text">
    <string>Refresh All Views</string>
   </property>
  </action>
  <action name="m_PlayAction">
   <property name="icon">
    <iconset>
     <normalon>:/AEngine_Editor/Resources/Images/button_black_play.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Play</string>
   </property>
   <property name="toolTip">
    <string>Play</string>
   </property>
  </action>
  <action name="m_PauseAction">
   <property name="icon">
    <iconset>
     <normalon>:/AEngine_Editor/Resources/Images/button_black_pause.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Pause</string>
   </property>
   <property name="toolTip">
    <string>Pause</string>
   </property>
  </action>
  <action name="m_StopAction">
   <property name="icon">
    <iconset>
     <normalon>:/AEngine_Editor/Resources/Images/button_black_stop.png</normalon>
    </iconset>
   </property>
   <property name="text">
    <string>Stop</string>
   </property>
  </action>
  <action name="m_StartPhysXDebuggerAction">
   <property name="text">
    <string>Starts PhysX Local Debugger</string>
   </property>
  </action>
  <action name="m_StopPhysXDebuggerAction">
   <property name="text">
    <string>Stop PhysX Local Debugger</string>
   </property>
  </action>
  <action name="m_MenuFileSaveProjectAction">
   <property name="text">
    <string>Save Project</string>
   </property>
  </action>
  <action name="m_MenuFileCreateProjectAction">
   <property name="text">
    <string>Create Project</string>
   </property>
  </action>
  <action name="m_MenuFileShowProjectAction">
   <property name="text">
    <string>Show Project</string>
   </property>
  </action>
  <action name="m_MenuFileLoadProjectAction">
   <property name="text">
    <string>Load Project</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RawGameAssetTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>Widgets\Raw Game Assets\RawGameAssetTreeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>GameAssetTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>Widgets\Game Assets\GameAssetTreeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>GameObjectTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>Widgets\Game Object\GameObjectTreeWidget.h</header>
  </customwidget>
  <customwidget>
   <class>GameObjectPropsTreeWidget</class>
   <extends>QTreeWidget</extends>
   <header>Widgets\Game Object\GameObjectPropsTreeWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../AEngine_editor.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>m_MenuFileExitAction</sender>
   <signal>triggered()</signal>
   <receiver>AEngine_EditorClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>639</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_GameAssetsViewAction</sender>
   <signal>triggered()</signal>
   <receiver>m_GameAssetsDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>513</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_RawGameAssetsViewAction</sender>
   <signal>triggered()</signal>
   <receiver>m_RawGameAssetsDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>91</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_GameObjectsViewAction</sender>
   <signal>triggered()</signal>
   <receiver>m_GameObjectsDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>903</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_GameObjectPropertiesViewAction</sender>
   <signal>triggered()</signal>
   <receiver>m_GameObjectPropsDock</receiver>
   <slot>show()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>845</x>
     <y>518</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
